### DISCUSSION TITLE
Starting HyperHDR with LED instance off by default?

### DISCUSSION THREAD

**Discussion by davethecripple** [awawa-dev/HyperHDR#627] [Q&A]
Hello all. I have a hopefully simple question I can't seem to figure out. I have HyperHDR and piccap working perfectly on my rooted LG OLED, however I've been trying to figure out if there is a way to have the LED instance default to turned OFF during startup rather than ON?

Primarily the TV is used to watch shows that do not benefit from ambient lighting, and rather than logging into the HyperHDR server and turn off the LED instance more often than not, I was curious if there is anyway to default the instance being off.

Thanks!

**Comment by games133** [comment]
ive currentll building my kit and ive added a smart switch in the power line to the power transformer. its a wifi switch so i can switch it on and off with that. then when i want the hyper come on i switch it on also i could tell google or alexa to turn it on and off. the system boots up and starts working. i belive there are timers as weill but not touched them. looking at the box in the pic. its the white part in the corner at a angle. once connected to wifi it all works great 
![20230820_181409 1](https://github.com/awawa-dev/HyperHDR/assets/141847398/6f0f1180-919d-4ae5-8657-c8da2a495aa8)

**Comment by davethecripple** [comment]
Shameless Bump - a separate device would be doable - but since I'm hosting HyperHDR on my rooted LG TV directly was hoping there is a software solution to start LED driver off by default

**Reply by jeremy-sidekicksolutions** [reply]
I'm doing the same thing with my LG Webos TV and using homeassistant. Not being able to find an option for this in HyperHDR, I think i'll have to set up an automation in home assistant to turn hyperHDR off whenever it's status changes to "on" when the previous state was "unavailable". There is a HyperHDR integration that can be added using HACS which provides access to a switch for the ambilight, and some of the other options. Not ideal as an option within HDR to start in an off state, but it is what it is. Hopefully they add that

**Comment by espilioto** [comment]
@awawa-dev Not opening a new issue as this is a very first world problem, but..

And I thought I was missing something major as I searched for this option for quite a while.
Having to turn off the instance till the screen capture is needed is very annoying, the leds might be doing something else (like LedFX) in the meanwhile.

**Comment by awawa-dev** [comment]
How do you imagine something like this would work? OK, we block at the "start" (as it turns out later, "start" is quite an imprecise term here) of the LEDs when running HyperHDR. So far it's simple. Maybe not entirely, because we assume that this applies to all instances or some specific ones? But then the just initialized grabber or some source from the flatbuffer calls in and orders the LEDs to be initialized. Are we blocking it too or maybe not? Because actually the "start" has already passed, but the grabber initialized for a few seconds and missed it. Or the flatbuffers responded only a few minutes after "start". Because I have the impression that in such a case no one will be completely satisfied and we may only break something here.

**Comment by espilioto** [comment]
Heh, makes sense that of course _it ain't that simple_ when explained the full chain of events.

Here's a second attempt at describing it (without actually hoping that it will get implemented as it does sound like something like this could impact the quality of life for most users/stability):

After all initialization all I do is click on the Disable Instance button, and since I only got 1 wled controller connected at the moment this gets executed:
`sendToHyperhdr('componentstate', '', '"componentstate":{"component":"ALL","state":false}')`

While it looks like all I want is to turn the leds off and that's not incorrect, the main bummer is that while HyperHDR is actively broadcasting to that instance other apps can't.

For now I've just disabled autostart.

**Comment by awawa-dev** [comment]
The problem is not that you can't turn off the LED at the beginning, but upcoming events wake them up anyway (grabber is initialized, flat buffer received a video frame, etc.).
If there is no video source, the current LED device will turn on briefly and then turn off and will not block any other application from accessing the LEDs or WLEDs. Why do you think this is a problem now? Do you have any video sources that prevent HyperHDR from turning off the LEDs? It's quite easy to add an option that will disable this short period of activity at HyperHDR start-up when the LEDs briefly illuminate, but this would still assume that your video sources are disabled or not present at the moment.

**Comment by awawa-dev** [comment]
`sendToHyperhdr('componentstate', '', '"componentstate":{"component":"ALL","state":false}')`

it is possible to add a feature that allows to perform such an action when HyperHDR starts. But this is unlikely to prevent a brief moment where HyperHDR will try to turn on the LED for a moment at the beginning.

**Comment by espilioto** [comment]
"A brief moment" is veeery far from a dealbreaker since I will gain back autostarting the service and also avoid having to visit the web interface every time my pc boots to disable the thing.

**Comment by awawa-dev** [comment]
Implemented in https://github.com/awawa-dev/HyperHDR/pull/737
I am asking you to test it because proper operation may depend on a specific configuration. The LEDs will not turn on even for a moment if the user has blocked their start. The feature is analogous to calling COMP_ALL = false from the JSON_API level

**Reply by espilioto** [reply]
I know it's a chore, but could I ask you to upload a build?
I'm on holiday and have no pc with me 😅

**Reply by awawa-dev** [reply]
Without PC it may be difficult. You need to download ZIP archive first from https://github.com/awawa-dev/HyperHDR/actions/runs/7573447093 extract it and install it

**Reply by espilioto** [reply]
Oh, you are right! Cicd to the rescue!
Gimme some time to forward this to someone closer to my setup than me and I'll get back to ya with the results.

Cheers!

**Comment by espilioto** [comment]
Ok, some preliminary testing shows that it seems to work fine.
Confirmed that the leds did not turn on even for a moment.
(did not test the disable when locked feature)

**Comment by awawa-dev** [comment]
Ok, thanks for the feedback. There will be some time for more thorough testing, because beta2 is planned first before the final version v20.

**Comment by espilioto** [comment]
Aye!
Will just keep this build until it gets merged.

Thanks again 😀

**Comment by mat-pandaz** [comment]
any updates on this? I use screen grab on my HTPC and it's automated to turn on the LEDs when they are needed and turn off when media finishes playing, however when i restart the machine it is on by default and I have to play then stop a piece of media to turn them off.  I'd prefer it if the video stream didn't auto enable the LEDs too.  When I stop playing HDR content the screen grab turns off for a moment, right when i automatically disable the lights, then the screen grab pics up the non HDR image again and automatically turns the LEDs on.  

Ultimately it would be nice to have an option to stop LED output automatically being switched on when the screen grab reports an image.

**Reply by awawa-dev** [reply]
Implemented long time ago, check v20 manifest

**Reply by mat-pandaz** [reply]
My bad, must have missed that setting thanks!
